Classes/Private/Manager/Video/OCTVideoEngine.m

```diff
@@ -125,22 +123,28 @@ - (BOOL)isSendingVideo
     return self.captureSession.isRunning;
 }
 
-- (CALayer *)videoCallPreview
+- (void)getVideoCallPreview:(void (^)(CALayer *))completionBlock
 {
     DDLogVerbose(@"%@: videoCallPreview", self);
-    AVCaptureVideoPreviewLayer *previewLayer = [AVCaptureVideoPreviewLayer layerWithSession:self.captureSession];
-
-    return previewLayer;
+    dispatch_async(self.processingQueue, ^{
+        AVCaptureVideoPreviewLayer *previewLayer = [AVCaptureVideoPreviewLayer layerWithSession:self.captureSession];
```

Comment:
Should layer be created on `processingQueue`?
- yes - can we dispatch_async again after creating and return it on main queue?
- no - no need for async?